subprojects/launcher/src/main/java/org/gradle/launcher/cli/converter/PropertiesToLogLevelConfigurationConverter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher.cli.converter;
+
+import org.gradle.api.logging.LogLevel;
+import org.gradle.api.logging.configuration.LoggingConfiguration;
+import org.gradle.launcher.daemon.configuration.GradleProperties;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class PropertiesToLogLevelConfigurationConverter {
+    public LoggingConfiguration convert(Map<String, String> properties, LoggingConfiguration loggingConfiguration) {
+        String logLevel = properties.get(GradleProperties.LOG_LEVEL_PROPERTY);
+        if (logLevel != null) {
+            LogLevel level = parseLogLevel(logLevel);
+            loggingConfiguration.setLogLevel(level);
+        }
+
+        return loggingConfiguration;
+    }
+
+    private LogLevel parseLogLevel(String value) {
+        try {
+            LogLevel logLevel = LogLevel.valueOf(value.toUpperCase());
+            if (logLevel == LogLevel.ERROR) {
+                throw new IllegalArgumentException("Log level cannot be set to ERROR.");
```

Comment:
The main exception will enumerate the valid possible values that can be set.  The causing exception will simply state you can't set it to ERROR.  I do agree that it does seem a little arbitrary to not allow ERROR.  @eriwen are you sure we don't want to allow it?